loadgen/generic_executor.go

```diff
@@ -9,35 +9,56 @@ import (
 	"go.uber.org/zap"
 )
 
-type GenericExecutor struct {
+type GenericExecutor[STORAGE any] struct {
+	// Used for Executor.PreScenario
+	PreScenarioImpl func(context.Context, ScenarioInfo, *STORAGE) error
 	// Function to execute a single iteration of this scenario
-	Execute func(ctx context.Context, run *Run) error
+	Execute func(context.Context, *Run, *STORAGE) error
+	// Used for Executor.PostScenario
+	PostScenarioImpl func(context.Context, ScenarioInfo, *STORAGE) error
 	// Default configuration if any.
 	DefaultConfiguration RunConfiguration
+	// Storage for the scenario
+	ScenarioStorage STORAGE
```

Comment:
Doesn't really seem to impose much burden. Places not using it just say `any` and can ignore everything else. For things that do need it, you get to avoid having to do ugly casts. Pre/post is a pretty common thing to want to do too.

If you'd rather have it be it's own new executor though, then fine I can just do that.

Reply:
Yeah, I think I probably will be inclined to use my own executor. We should consider, for those wanting their own executor, helpers for executing iterations/durations.

Reply:
undone